src/Symfony/Component/Console/Parallelization/Parallelization.php

```diff
@@ -0,0 +1,475 @@
+     * left to process, new child processes are spawned automatically.
+     *
+     * @param InputInterface  $input  The console input
+     * @param OutputInterface $output The console output
+     */
+    protected function executeMainProcess(InputInterface $input, OutputInterface $output)
+    {
+        $this->runBeforeFirstCommand($input, $output);
+
+        $numberOfProcesses = (int) $input->getOption('processes');
+        $elements = $input->getArgument('element') ? [$input->getArgument('element')] : $this->fetchElements($input);
+        $count = count($elements);
+        $segmentSize = 1 === $numberOfProcesses ? $count : $this->getSegmentSize();
+        $batchSize = $this->getBatchSize();
+        $rounds = 1 === $numberOfProcesses ? 1 : ceil($count * 1.0 / $segmentSize);
+        $batches = ceil($segmentSize * 1.0 / $batchSize) * $rounds;
+
+        if (0 === $numberOfProcesses) {
+            throw new InvalidArgumentException(sprintf('Requires at least one process, "%s" given.', $input->getOption('processes')));
+        }
+
+        $output->writeln(sprintf(
+            'Processing %d %s in segments of %d, batches of %d, %d %s, %d %s in %d %s',
+            $count,
+            $this->getElementName($count),
+            $segmentSize,
+            $batchSize,
+            $rounds,
+            1 === $rounds ? 'round' : 'rounds',
+            $batches,
+            1 === $batches ? 'batch' : 'batches',
+            $numberOfProcesses,
+            1 === $numberOfProcesses ? 'process' : 'processes'
+        ));
+        $output->writeln('');
+
+        $progressBar = new ProgressBar($output, $count);
+        $progressBar->setFormat('debug');
+        $progressBar->start();
+
+        if ($count <= $segmentSize || 1 === $numberOfProcesses) {
+            $i = 0;
+
+            // Run in main process if we have only one segment
+            foreach ($elements as $element) {
+                if (0 === $i) {
+                    $this->runBeforeBatch($input, $output);
+                }
+
+                $this->runTolerantSingleCommand($element, $input, $output);
+
+                $progressBar->advance();
+                ++$i;
+
+                if ($i >= $this->getBatchSize()) {
+                    $this->runAfterBatch($input, $output);
+                    $i = 0;
+                }
+            }
+
+            if (0 !== $i) {
+                $this->runAfterBatch($input, $output);
+            }
+        } else {
+            // Distribute if we have multiple segments
+            $commandTemplate = sprintf('%s bin/console %s %s --child --env=%s --verbose --no-debug',
+                self::detectPhpExecutable(),
+                $this->getName(),
+                implode(' ', array_slice($input->getArguments(), 1)),
+                $input->getOption('env')
+            );
+            $terminalWidth = current($this->getApplication()->getTerminalDimensions());
```

Comment:
Application::getTerminalDimensions() has been deprecated since version 3.2 and removed since 4.0